TicTacToe/TicTacToe/ComputerLogic.cs

```diff
@@ -29,59 +51,49 @@ public class ComputerLogic {
+                score = 0;
             }
-            return move;
+            return score;
         }
 
-        private Moves GetBestMove(string[] gameBoard, string marker) {
+        private int GetScore(string[] gameBoard, int alpha, int beta, string marker, int depth) {
             List<int> availableSpaces = GetAvailableSpaces(gameBoard);
-
+            
             if (this.winConditions.IsWinner(gameBoard) || availableSpaces.Count == 0) {
-                Moves move = new Moves();
-                move = GetScore(gameBoard, marker, move);
-                return move;
-            }
-
-            List<Moves> moves = new List<Moves>();
-
-            for (int i = 0; i < availableSpaces.Count; i++) {
-                Moves moveValues = new Moves();
-                moveValues.Spot = availableSpaces.ElementAt(i);
-                string[] gameBoardCopy = (string[]) gameBoard.Clone();
-                gameBoardCopy[availableSpaces.ElementAt(i)] = marker;
-                int score = GetBestMove(gameBoardCopy, AlternateMarker(marker)).Score;
-                moveValues.Score = score;
-                moves.Add(moveValues);
-            }
-
-            int bestMove = 0;
+                return GetGameScore(gameBoard, AlternateMarker(marker), depth);
+            } 
+            
             if (marker == "O") {
-                int bestMoveScore = -1000;
-                for (int i = 0; i < moves.Count; i++) {
-                    if (moves[i].Score > bestMoveScore) {
-                        bestMoveScore = moves[i].Score;
-                        bestMove = i;
-                    }
+                for (int i = 0; i < availableSpaces.Count; i++) {
+                    string[] gameBoardCopy = (string[]) gameBoard.Clone();
+                    gameBoardCopy[availableSpaces.ElementAt(i)] = marker;
+                    int score = GetScore(gameBoardCopy, alpha, beta, AlternateMarker(marker), depth++);
+                    alpha = Math.Max(alpha, score );
+                    if (beta <= alpha || depth == 15) {
+                        break;
+                    }  
                 }
+                return alpha;
             } else {
-                int bestMoveScore = 1000;
-                for (int i = 0; i < moves.Count; i++) {
-                    if (moves[i].Score < bestMoveScore) {
-                        bestMoveScore = moves[i].Score;
-                        bestMove = i;
-                    }
+                for (int i = 0; i < availableSpaces.Count; i++) {
+                    string[] gameBoardCopy = (string[]) gameBoard.Clone();
+                    gameBoardCopy[availableSpaces.ElementAt(i)] = marker;
+                    int score = GetScore(gameBoardCopy, alpha, beta, AlternateMarker(marker), depth++);
+                    beta = Math.Min(beta, score);
+                    if (beta <= alpha || depth == 15) {
+                        break;
+                    }  
                 }
+                return beta;
```

Comment:
How could you eliminate the duplication between the bodies of the `if` and `else`?

Also, there's that `15` again.